File: test/client_code/tests/server/echo.js
/*
server/echo.js
*/

export default async () => {
  const { frame } = await use("@/frame/");
  const { server } = await use("@/server");
  const { Exception, match } = await use("@/rollo/");
  const dto = [
    { yes: true, no: false, number: 1, text: crypto.randomUUID() },
    [crypto.randomUUID(), 0, 1, true, false, null],
  ];
  const { result: _dto, meta } = await server.echo(dto[0], ...dto[1]);
  Exception.if(!match(dto, _dto), `Server connection could not be verified.`);
  console.info("Server connection verified for:", meta);
};

File: test/client_code/tests/session/echo.py
"""
session/echo.py
"""


def main(use):

    


    echo = use("rpc/echo", dict(spinner=True))
    response = echo(42)
    ##result = response.get("result")
    ##print("result:", result)
    meta = response.get("meta")
    print("meta:", meta)

    server = use("@/server").server
    response = server.echo(42)
    ##result = response.result
    ##print("result:", result)
    meta = response.meta
    print("meta:", meta)
